apm-sniffer/optional-plugins/optional-spring-plugins/spring-webflux-5.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/spring/webflux/v5/reactive/DispatcherHandlerHandleMethodInterceptor.java

```diff
@@ -16,62 +16,93 @@
 import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
 import org.springframework.web.server.ServerWebExchange;
 import org.springframework.web.server.ServerWebExchangeDecorator;
 import org.springframework.web.server.adapter.DefaultServerWebExchange;
+import reactor.core.publisher.Mono;
 
-import java.lang.reflect.Method;
-import java.util.List;
-
+/**
+ * @author Born
+ */
 public class DispatcherHandlerHandleMethodInterceptor implements InstanceMethodsAroundInterceptor {
 
-    private static final String WIP_OPERATION_NAME = "WEBFLUX.handle";
+    private ContextManagerExtendService extendService = ServiceManager.INSTANCE.findService(ContextManagerExtendService.class);
 
     @Override
     public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
                              MethodInterceptResult result) throws Throwable {
-        EnhancedInstance instance = DispatcherHandlerHandleMethodInterceptor.getInstance(allArguments[0]);
-        if (instance != null) {
-            ContextCarrier contextCarrier = new ContextCarrier();
-            CarrierItem next = contextCarrier.items();
-            ServerWebExchange exchange = (ServerWebExchange) allArguments[0];
-            HttpHeaders headers = exchange.getRequest().getHeaders();
-            while (next.hasNext()) {
-                next = next.next();
-                List<String> header = headers.get(next.getHeadKey());
-                if (header != null && header.size() > 0) {
-                    next.setHeadValue(header.get(0));
-                }
-            }
-            AbstractSpan span = ContextManager.createEntrySpan(WIP_OPERATION_NAME, contextCarrier);
-            span.setComponent(ComponentsDefine.SPRING_WEBFLUX);
-            SpanLayer.asHttp(span);
-            Tags.URL.set(span, exchange.getRequest().getURI().toString());
-            instance.setSkyWalkingDynamicField(span);
-        }
+
     }
 
     @Override
     public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
                               Object ret) throws Throwable {
-        return ret;
+        EnhancedInstance instance = getInstance(allArguments[0]);
+
+        ServerWebExchange exchange = (ServerWebExchange) allArguments[0];
+
+        ContextCarrier carrier = new ContextCarrier();
+        CarrierItem next = carrier.items();
+        HttpHeaders headers = exchange.getRequest().getHeaders();
+        while (next.hasNext()) {
+            next = next.next();
+            List<String> header = headers.get(next.getHeadKey());
+            if (header != null && header.size() > 0) {
+                next.setHeadValue(header.get(0));
+            }
+        }
+
+        String pathStr = exchange.getRequest().getPath().value();
+        String methodStr = exchange.getRequest().getMethodValue();
```

Comment:
if the route is `/test/{var}`, when one request url is `/test/a`, the methodStr is `/test/a`,right?
this is why before plugin, `DispatcherHandlerInvokeHandlerMethodInterceptor ` reset operationName.
in this case the `EndpointNameDictionary` will growth more, so if not support this case, need to explain on `the md file` to everyone know. FYI @wu-sheng

Reply:
you're right,  but this problem exists in web interfaces such as tomcat and undertow. right? Is it possible to have a way. When there is a lot of data, the collection program can combine characteristics similarly.

Reply:
I think this should be supported. In the Gateway related plugin, this is super important.

Reply:
What should I do? Keep using old operationName？

Reply:
Make `/test/{var}` as the operation name, rather than `/test/a`, `/test/b`. The real URL only exists in TAG only.

Reply:
I do n’t have a good way, can anyone help. @wu-sheng  @zhaoyuguang

Reply:
If the spring-web supported by tomcat or undertow should be no problem, if you have this case, please add a issue to others know, thanks.
 
previous webflux plugins didn't write very well at performance, but can work,  you can see `apm-sniffer/apm-sdk-plugin/spring-plugins/webflux-5.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/spring/webflux/v5/DispatcherHandlerHandleResultMethodInterceptor.java`

Reply:
all operationName  use "WEBFLUX.handle"? right?

Reply:
https://github.com/apache/skywalking/blob/52de0c6dfad5750239267d7808a5016eb51a5068/apm-sniffer/apm-sdk-plugin/tomcat-7.x-8.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/tomcat78x/TomcatInvokeInterceptor.java#L75   @zhaoyuguang  if tomcat servlet url-pattern is ``/**``  what's going to happen

Reply:
This will be overridden by SpringMVC plugin if it is activated. We are using Spring Controller pattern today.